### ISSUE TITLE
weird results while running the example code

### ISSUE THREAD

**Issue by bhomaidan1990** [IDA-TUBS/pycpa#22]
1. When I run the Tutorial code I get: `T22 = q x WCET: 2 x 9 =18` which is wrong! 
While in your explanation of the [tutorial](https://pycpa.readthedocs.io/en/latest/tutorial.html#analysis) you write 
```py
T22: wcrt=11 b_wcrt=T21:eta x WCET: 1 x 2=2, blocker:0, q x WCET: 1 x 9=9
```
which is correct!
**q** value should be **q=1**, can you please fix that?

2. Another point is that in [end-to-end-path-latency-analysis](https://pycpa.readthedocs.io/en/latest/tutorial.html#end-to-end-path-latency-analysis) :
```py
# perform path analysis of selected paths
    for p in paths:
```
paths is not defined! 
can you please add `paths = [p1, p2]` before that, or anything similar to that, so the code can run with beginners running the code step by step like me? thanks in advance

**Comment by jschlatow** [comment]
@bhomaidan1990 are you referring to the [SPP example](https://pycpa.readthedocs.io/en/latest/spp_example.html) or the [tutorial](https://pycpa.readthedocs.io/en/latest/tutorial.html)?
For the tutorial (step 1) you should get the following results:
```
T21: wcrt=11
    b_wcrt=blocker:9, q*WCET:1*2=2
T22: wcrt=18
    b_wcrt=T21:eta*WCET:1*2=2, blocker:0, q*WCET:2*9=18
T11: wcrt=10
    b_wcrt=q*WCET:1*10=10
T12: wcrt=13
    b_wcrt=T11:eta*WCET:1*10=10, q*WCET:1*3=3
T31: wcrt=11
    b_wcrt=T32:eta*WCET:2*3=6, q*WCET:1*5=5
T32: wcrt=3
    b_wcrt=q*WCET:1*3=3
```

Can you confirm that you get the same results by running `python examples/tutorial.py --steps step1`?

I will , of course, update the tutorial with the correct results and also add your suggestion regarding the definition of `paths`.

**Comment by bhomaidan1990** [comment]
I'm referring to the tutorial, and yes I'm getting this same results, but this result `T22: wcrt=18 ..... q = 2` is **not correct**, q should be 1, and the `T22 wcrt` should be `wcrt=9`

**Comment by jschlatow** [comment]
To explain the result, we must have a look at the input event model of T22. I uploaded the file [event-model-T22plot.pdf](https://github.com/IDA-TUBS/pycpa/files/5791335/event-model-T22plot.pdf), which shows a plot of the event model at the end of the analysis. The important curve here is delta-.
The busy-window analysis for T22 starts with q=1. T22 has a worst-case execution time (WCET) of 9 and, since it executes on a lower priority than T21, is interferred by the latter. Thus, the q-event busy period is 11 for q=1. Yet, looking at delta-(q+1=2), we see that there is the possibility that another event arrives within less than 11 time units. Therefore, we also have to compute the 2-event busy period. In short, this explains why you see the q=2 in the results. The worst-case response time (WCRT) is then computed by  comparing the arrival time of each event with the corresponding q-event busy time. The first event would have a response time of 11 whereas the second event has a response time of 18 since it arrives at least two time units later than the first event and the 2-event busy time is 20 as you see in the posted results.

The input event model of T22 is explained as follows: The input of T12 has a period of 15 and jitter of 1, i.e. we can see a minimum distance of 14. Yet, as T12 has a lower priority than T11 it has a relatively large WCRT of 13 but a small best-case execution time of 1. In consequence, it adds a jitter of WCRT-BCET=12 to the original input event model. A period of 15 and jitter of 13 leads to the minimum distance of 2 that we saw before. (Note, that pycpa does some more tricks to improve the propagates event models for n>2. This is why the plot from above does not exactly follow a periodic with jitter event model.)
